### ISSUE TITLE
Poll: breaking API change?

### ISSUE THREAD

**Issue by jamesplease** [jamesplease/react-composer#43]
**To cast your vote, please leave a reaction emoji of :+1: or :-1: to this issue!**

---

@erikthedeveloper proposed a really cool API for Composer over in #39 . The change is this: instead of `components` callbacks receiving `results`, they receive `({ results, render })`. You can then manually specify the `render` function as the render prop of the component.

Here is @erikthedeveloper 's example from that PR, showing it in use:

```jsx
<Composer
  components={[
    // React elements may be passed for simple/basic use cases.
    // Assumes single value produced such as `props.children(producedValue)`
    <Echo value="outer" />,

    // OR A function may be passed to produce an element.

    // Utilizing outer results for inner components
    ({ render, results: [outerResult] }) => (
      <Echo value={`${outerResult.value} + middle`} children={render} />
    ),
    
    // Differing render prop signature (multi-arg producers)
    ({ /* results, */ render }) => (
      <DoubleEcho value="spaghetti">
        {(one, two) => render([one, two])}
      </DoubleEcho>
    ),
    
    // Differing render prop names...
    ({ /* results, */ render }) => (
      <EchoRenderProp value="spaghetti" renderProp={render} />
    )
    ]}
  ]}
  children={/* ... */}
/>
```

### Negatives

- This is a breaking API change.

### Positives

- Reduces API surface area. `renderPropName` and `mapResult` are no longer needed, and no functionality is lost
- Provides a good solution to the issue where you would get console warnings for omitting the render prop ( #42 )
- Supports more easily composing render prop components with different render prop names ( #37 )

---

tl;dr, it is a flexible, small, and powerful API that is a big breaking change from what we currently have. What do people think? Should it be changed?

Add your vote to the 👍 / 👎 tally below!

**Comment by jamesplease** [comment]
Thanks everyone for sharing your opinions here! It looks like most people are in favor of this change, so...it's gonna happen!

**Comment by jamesplease** [comment]
I am closing this issue because we've decided to go ahead and merge #39 . Thanks everyone for your contributions to the poll! The new version will be released in a day or two.
